etcdserver/server.go

```diff
@@ -458,11 +458,16 @@ func NewServer(cfg *ServerConfig) (srv *EtcdServer, err error) {
 		}
 	}
 	srv.consistIndex.setConsistentIndex(srv.kv.ConsistentIndex())
-
-	srv.authStore = auth.NewAuthStore(srv.be,
+	tp := auth.NewTokenProvider(cfg.TokenOpts,
 		func(index uint64) <-chan struct{} {
 			return srv.applyWait.Wait(index)
-		})
+		},
+	)
+	if tp == nil {
+		plog.Errorf("failed to create token provider, required token type is %s", cfg.TokenType)
+		return nil, auth.ErrInvalidAuthOpts
```

Comment:
probably better to return an error from `NewTokenProvider` instead of having etcdserver being responsible for choosing auth errors?